### PR TITLE
test: fixed postcommentsview test cases

### PR DESCRIPTION
**Pull request by ayesha-waris**
[INF-926](https://2u-internal.atlassian.net/browse/INF-926)
### Description

Some test cases were failing due to structure of comments factory used in PostCommentsView Tests.
This ticket intends to add those removed test cases and fix comments factory.

### CODE REVIEW

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 1]
```diff
@@ -201,6 +222,16 @@ describe('ThreadView', () => {
       expect(JSON.parse(axiosMock.history.patch[axiosMock.history.patch.length - 1].data)).toMatchObject(data);
     }
 
+    it('should not allow posting a comment on a closed post', async () => {
+      axiosMock.reset();
+      await mockAxiosReturnPagedComments(closedPostId, true);
+      await waitFor(() => renderComponent(closedPostId, true));
+      const comment = await waitFor(() => screen.findAllByTestId('comment-comment-4'));
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:229]: rename it to "comments"

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 2]
```diff
@@ -491,8 +550,10 @@ describe('ThreadView', () => {
   });
 
   describe('for discussion thread', () => {
+    const findLoadMoreCommentsButton = () => screen.findByTestId('load-more-comments');
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:553]: rename findLoadMoreCommentsButton to "loadMoreCommentsButton"

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 3]
```diff
@@ -491,8 +550,10 @@ describe('ThreadView', () => {
...
+
     it('shown post not found when post id does not belong to course', async () => {
-      renderComponent('unloaded-id');
+      await waitFor(() => renderComponent('unloaded-id'));
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:556]: I think we don't need waitFor here. recheck it please
await renderComponent('unloaded-id');

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 4]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
         .not
         .toBeInTheDocument();
     });
+
+    it('pressing load more button will load next page of comments', async () => {
+      await waitFor(() => renderComponent(discussionPostId));
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:571]: I think we don't need waitFor here. recheck it please
await renderComponent(discussionPostId);

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 5]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
...
+      await mockAxiosReturnPagedComments(discussionPostId, false, 2);
+
+      const loadMoreButton = await findLoadMoreCommentsButton();
+      fireEvent.click(loadMoreButton);
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:575]: wrap this in act. The act function ensures that all updates triggered by the event are processed and any resulting asynchronous behavior is properly handled. so it is recommended to wrap fireEvent.click with the act.
await act(async () => {
  fireEvent.click(loadMoreButton);
});

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 6]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
...
+      await screen.findByTestId('comment-comment-1');
+      await screen.findByTestId('comment-comment-4');
+    });
+
+    it('newly loaded comments are appended to the old ones', async () => {
+      await waitFor(() => renderComponent(discussionPostId));
+      await mockAxiosReturnPagedComments(discussionPostId, false, 2);
+
+      const loadMoreButton = await findLoadMoreCommentsButton();
+      fireEvent.click(loadMoreButton);
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:586]: same as above
await act(async () => {
fireEvent.click(loadMoreButton);
});

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 7]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
...
+
+      await screen.findByTestId('comment-comment-1');
+      // check that comments from the first page are also displayed
+      expect(screen.queryByTestId('comment-comment-4'))
+        .toBeInTheDocument();
+    });
+  });
+
+  describe('for question thread', () => {
+    const findLoadMoreCommentsButtons = () => screen.findByTestId('load-more-comments');
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:596]: rename it to "loadMoreCommentsButton"

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 8]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
...
+
+      const loadMoreButton = await findLoadMoreCommentsButton();
+      fireEvent.click(loadMoreButton);
+
+      await screen.findByTestId('comment-comment-1');
+      await screen.findByTestId('comment-comment-4');
+    });
+
+    it('newly loaded comments are appended to the old ones', async () => {
+      await waitFor(() => renderComponent(discussionPostId));
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:582]: I think we don't need waitFor here. recheck it please
await renderComponent(discussionPostId);

File: src/discussions/post-comments/PostCommentsView.test.jsx [hunk 9]
```diff
@@ -505,6 +566,66 @@ describe('ThreadView', () => {
...
+      expect(await screen.findByTestId('comment-comment-4'))
+        .toBeInTheDocument();
+      expect(screen.queryByTestId('comment-comment-5'))
+        .not
+        .toBeInTheDocument();
+    });
+
+    it('pressing load more button will load next page of comments', async () => {
+      await mockAxiosReturnPagedComments(questionPostId);
+      await waitFor(() => renderComponent(questionPostId));
```
Comment (sundasnoreen12) [inline src/discussions/post-comments/PostCommentsView.test.jsx:611]: same as above

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]

File: src/discussions/post-comments/PostCommentsView.test.jsx (hunk 3 above)
Comment (ayesha-waris) [reply src/discussions/post-comments/PostCommentsView.test.jsx:556]: without waitFor test case fails in PR

File: src/discussions/post-comments/PostCommentsView.test.jsx (hunk 4 above)
Comment (ayesha-waris) [reply src/discussions/post-comments/PostCommentsView.test.jsx:571]: without waitFor test case fails in PR

File: src/discussions/post-comments/PostCommentsView.test.jsx (hunk 2 above)
Comment (ayesha-waris) [reply src/discussions/post-comments/PostCommentsView.test.jsx:553]: I think with find, the methods represents it functionality

**Review by sundasnoreen12 — APPROVED** [review]

**Pull request merged**